src/io/memory.cr

```diff
@@ -182,7 +182,7 @@ class IO::Memory < IO
   end
 
   # :nodoc:
-  def skip(bytes_count) : Nil
+  def skip(bytes_count : Int32 | UInt32) : Nil
```

Comment:
Where does this union type come from? It smells incorrect to me.

Reply:
This method overrides `IO#skip(bytes_count : Int) : Nil`, so the type restriction should be equivalent.

Perhaps the typer tool could take restrictions from super defs into account? 🤔

Reply:
Aha! Not a bad idea, though detecting super defs is non-trivial. I think in this situation, because this particular def had a more expansive (specifically: none) type restriction than its parent, the compiler treats it as a separate def rather than a child def. 

At least, that's what I think is happening :sweat_smile: Based on [DefWithMetadata#compare_strictness](https://github.com/crystal-lang/crystal/blob/master/src/compiler/crystal/semantic/restrictions.cr#L69) method and comments around it.

Reply:
At least for this particular instance, I can update this to use `Int` instead

Reply:
`#compare_strictness` is only about ordering defs to figure out which is the closest matching overload.

For digging up super defs, you can take a look at `Call#lookup_super_matches` which handles that for the `super` keyword.